fearless/Modules/Staking/StakingPayoutConfirmation/StakingPayoutConfirmationPresenter.swift

```diff
@@ -139,96 +89,167 @@ extension StakingPayoutConfirmationPresenter: StakingPayoutConfirmationPresenter
+            }
+
+        case let .failure(error):
+            logger?.error("Account Info subscription error: \(error)")
         }
+    }
 
-        provideFee()
+    func didReceiveStashItem(result: Result<StashItem?, Error>) {
+        switch result {
+        case let .success(stashItem):
+            self.stashItem = stashItem
+
+        case let .failure(error):
+            logger?.error("Stash subscription item error: \(error)")
+        }
     }
 
-    func didReceive(feeError: Error) {
-        handle(error: feeError)
+    func didReceivePayee(result: Result<RewardDestinationArg?, Error>) {
+        switch result {
+        case let .success(payee):
+            guard let payee = payee, let stashItem = stashItem
+            else {
+                rewardDestination = nil
+                return
+            }
+
+            do {
+                let rewardDestination = try RewardDestination(
+                    payee: payee,
+                    stashItem: stashItem,
+                    chain: chain
+                )
+
+                switch rewardDestination {
+                case .restake:
+                    self.rewardDestination = .restake
+                    provideViewModel()
+
+                case let .payout(payoutAddress):
+                    interactor.provideRewardDestination(for: payoutAddress)
+                }
+            } catch {
+                logger?.error("Did receive payee item error: \(error)")
+            }
+        case let .failure(error):
+            logger?.error("Payee subscription error: \(error)")
+        }
     }
 
-    func didReceive(balance: AccountData?) {
-        if let availableValue = balance?.available {
-            self.balance = Decimal.fromSubstrateAmount(
-                availableValue,
-                precision: asset.precision
-            )
-        } else {
-            self.balance = 0.0
+    func didReceiveRewardDestiination(result: Result<RewardDestination<DisplayAddress>?, Error>) {
+        switch result {
+        case let .success(rewardDestination):
+            self.rewardDestination = rewardDestination
+            provideViewModel()
+        case let .failure(error):
+            logger?.error("Did receive reward destination error: \(error)")
         }
     }
 
-    func didReceive(balanceError: Error) {
-        handle(error: balanceError)
+    func didStartPayout() {
+        view?.didStartLoading()
     }
 
-    func didReceive(price: PriceData?) {
-        priceData = price
-        provideFee()
-        provideViewModel()
+    func didCompletePayout(txHashes: [String]) {
+        txHashes.forEach { txHash in
+            logger?.info("Did send payouts: \(txHash)")
+        }
+
+        view?.didStopLoading()
+
+        wireframe.complete(from: view)
     }
 
-    func didReceive(priceError: Error) {
-        handle(error: priceError)
+    func didFailPayout(error: Error) {
+        view?.didStopLoading()
+
+        handle(error: error)
```

Comment:
I think we don't show an error in handle method if it is unknown. Please, show CommonError.undefined instead of just logging